### PR TITLE
[aboot] use ram partition for /var/log for devices with 3.7G disks

### PR DESCRIPTION
**Pull request by yxieca**
#### Why I did it
Master/202012 image size grew quite a bit. 3.7G harddrive can no longer hold one image and safely upgrade to another image. Every bit of harddrive space is precious to save now.

Also sh syntax seemingly changed, [ condition ] && action was a legit syntax in 201911 branch but it is an error when condition not met with 202012 or later images. Change the syntax to if statement to avoid the issue.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [x] 202106

### CODE REVIEW

**Review by lguohan — DISMISSED** [review]

**Review by Staphylo — DISMISSED** [review]
Looks good to me, thanks Ying!

File: files/initramfs-tools/union-mount.j2 [hunk 1]
```diff
@@ -151,16 +165,21 @@ mount --bind ${rootmnt}/host/$image_dir/boot ${rootmnt}/boot
...
+    mount -t tmpfs -o rw,nosuid,nodev,size=${varlogsize}M tmpfs ${rootmnt}/var/log
+    if [ -f ${rootmnt}/host/disk-img/var-log.ext4 ]; then
+        rm -rf ${rootmnt}/host/disk-img/var-log.ext4
+    fi
 else
-  [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && fsck.ext4 -v -p ${rootmnt}/host/disk-img/var-log.ext4 2>&1 \
-                                                  | gzip -c >> /tmp/fsck.log.gz
-  [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && mount -t ext4 -o loop,rw ${rootmnt}/host/disk-img/var-log.ext4 ${rootmnt}/var/log
+    if [ -f ${rootmnt}/host/disk-img/var-log.ext4 ]; then
+        fsck.ext4 -v -p ${rootmnt}/host/disk-img/var-log.ext4 2>&1 | gzip -c >> /tmp/fsck.log.gz
```
Comment (Staphylo) [inline files/initramfs-tools/union-mount.j2:178]: Couldn't these 2 statements (`fsck` & `mount`) fit in a single `if` condition?
I don't believe `fsck.ext4` would remove the blob.
However if you prefer keeping it this way for now I'm fine with it.

File: files/initramfs-tools/union-mount.j2 [hunk 2]
```diff
@@ -151,16 +165,21 @@ mount --bind ${rootmnt}/host/$image_dir/boot ${rootmnt}/boot
 
 ## Mount loop device or tmpfs for /var/log
 if $logs_inram; then
-  # NOTE: some platforms, when reaching initramfs stage, have a small
-  #       limit of mounting tmpfs partition, potentially due to amount
-  #       of RAM available in this stage. e.g. Arista 7050-qx32[s] and 7060-cx32s
-  set_tmpfs_log_partition_size
-  mount -t tmpfs -o rw,nosuid,nodev,size=${varlogsize}M tmpfs ${rootmnt}/var/log
-  [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && rm -rf ${rootmnt}/host/disk-img/var-log.ext4
+    # NOTE: some platforms, when reaching initramfs stage, have a small
```
Comment (Staphylo) [inline files/initramfs-tools/union-mount.j2:168]: Reading this comment I feels it should become
```
NOTE: Some platforms have a small flash and therefore the log partition takes valuable space.
      To improve the longevity of these devices storing the logs in memory will permit more
      SONiC image growth before being bottlenecked.
```

**Review by Staphylo — APPROVED** [review]
Thanks Ying for figuring that one out!
The `set -e` in the shebang was pretty tricky indeed.

**Review by saiarcot895 — APPROVED** [review]

File: files/initramfs-tools/union-mount.j2 (hunk 1 above)
Comment (yxieca) [reply files/initramfs-tools/union-mount.j2:184]: Good catch. Thanks!

File: files/initramfs-tools/union-mount.j2 (hunk 2 above)
Comment (yxieca) [reply files/initramfs-tools/union-mount.j2:168]: The statement means that we want to limit the size we give to ramdisk. I made it clearer. I also added your statement.

**Review by Staphylo — DISMISSED** [review]
LGTM

**Review by Staphylo — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Pull request merged**